### PR TITLE
feat: updates `useSelection` implementation to better support all components

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. Makes `useSelection` API more generic to support `@fluentui/react-combobox` use cases
2. updates `@fluentui/react-table` to work with modifications
3. modifies `@fluentui/react-combobox` to adopt `useSelection` API from `react-utilities`

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/09765cad9ee0868a3c157b90056d2e52f3e59855'>09765cad9ee0868a3c157b90056d2e52f3e59855</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Avatar-mount_base.html">592</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Avatar-mount.html">594</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Button-mount_base.html">306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Button-mount.html">292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Field-mount_base.html">1062</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Field-mount.html">1050</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProvider-mount_base.html">655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProvider-mount.html">667</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-mount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">62</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">74</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/MakeStyles-mount_base.html">852</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/MakeStyles-mount.html">872</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Persona-mount_base.html">1659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/Persona-mount.html">1609</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/SpinButton-mount_base.html">1339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28380/perf-test-react-components/SpinButton-mount.html">1320</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-combobox/src/utils/useSelection.ts [hunk 1]
```diff
@@ -1,49 +1,38 @@
...
-          newSelection = [...selectedOptions, option.value];
-        }
-      }
-
-      setSelectedOptions(newSelection);
-      onOptionSelect?.(event, { optionValue: option.value, optionText: option.text, selectedOptions: newSelection });
-    },
-    [onOptionSelect, multiselect, selectedOptions, setSelectedOptions],
-  );
+  const selectOption = useEventCallback((event: SelectionEvents, option: OptionValue) => {
```
Comment (ling1726) [inline packages/react-components/react-combobox/src/utils/useSelection.ts:15]:
```suggestion
  const toggleOption = useEventCallback((event: SelectionEvents, option: OptionValue) => {
```

Looking at the previous code this should probably have always been called toggle

**Review by ling1726 — APPROVED** [review]

**Pull request closed without merging**